### PR TITLE
Port bay newest loadout system

### PR DESCRIPTION
**Pull request by Alberyk**
-ports bay newest loadout system, which is pretty much a tab of their own, with different sections instead of a single list
-adds code that allows for several tweaks of custom loadout, like color, item path and contents
-adds dumb lunchboxes that can be filled with some food selection
-adds the option to select pre-filled vacuum-flasks or flasks
-also tries to fix the issues with xeno gear again

### CODE REVIEW

File: code/modules/client/preference_setup/general/04_equipment.dm [hunk 1]
```diff
@@ -7,14 +7,12 @@
...
 
 /datum/category_item/player_setup_item/general/equipment/save_character(var/savefile/S)
 	S["underwear"]	<< pref.underwear
 	S["undershirt"]	<< pref.undershirt
 	S["socks"]		<< pref.socks
 	S["backbag"]	<< pref.backbag
-	S["gear"]		<< pref.gear
 
 /datum/category_item/player_setup_item/general/equipment/gather_load_query()
 	return list("ss13_characters" = list("vars" = list("underwear", "undershirt", "socks", "backbag", "gear"), "args" = list("id")))
```
Comment (skull132) [inline code/modules/client/preference_setup/general/04_equipment.dm]: You should move gear out from the SQL system here, and into the loadout/loadout.dm. Procs you have to update:
* gather_load_query()
* gather_save_query()
* gather_save_parameters()
* sanitize_character().

Comment (Alberyk) [inline code/modules/client/preference_setup/general/04_equipment.dm]: Changes probably done.

File: code/modules/client/preference_setup/general/04_equipment.dm [hunk 2]
```diff
@@ -7,31 +7,28 @@
...
 /datum/category_item/player_setup_item/general/equipment/gather_save_parameters()
-	return list(":underwear" = pref.underwear, ":undershirt" = pref.undershirt, ":socks" = pref.socks, ":backbag" = pref.backbag, ":gear" = list2params(pref.gear), ":id" = pref.current_character, ":ckey" = pref.client.ckey)
+	return list(":underwear" = pref.underwear, ":undershirt" = pref.undershirt, ":socks" = pref.socks, ":backbag" = pref.backbag, ":id" = pref.current_character, ":ckey" = pref.client.ckey)
 
 /datum/category_item/player_setup_item/general/equipment/sanitize_character(var/sql_load = 0)
 	if (sql_load)
 		pref.backbag = text2num(pref.backbag)
-		pref.gear	= params2list(pref.gear)
 
 	pref.backbag	= sanitize_integer(pref.backbag, 1, backbaglist.len, initial(pref.backbag))
```
Comment (skull132) [inline code/modules/client/preference_setup/general/04_equipment.dm]: Line 35 & 36 should also be removed. They're handled in the other .dm file.

this thing:
```
    if (!islist(pref.gear))
        pref.gear = list()
```

Comment (Alberyk) [inline code/modules/client/preference_setup/general/04_equipment.dm]: Removed now.

**Pull request merged**